config/routes.rb

```diff
@@ -1,5 +1,11 @@
 Rails.application.routes.draw do
-  root "sessions#index"
+  constraints AuthenticatedConstraint.new do
+    root "sessions#index"
+  end
+
+  constraints UnauthenticatedConstraint.new do
+    root "user_sessions#new", as: :unauthenticated_root
+  end
```

Comment:
Constraints allow us to have two different root paths. https://api.rubyonrails.org/classes/ActionDispatch/Routing/Mapper/Scoping.html#method-i-constraints